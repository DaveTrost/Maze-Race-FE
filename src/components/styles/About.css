@font-face {
  font-family: "joystix";
  src: url('../../assets/fonts/joystix.ttf')
  }

.AboutElements {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  

  h2 {
    font-size: 4em;
    font-family: "KARNIVOT";
    background: -webkit-linear-gradient(bottom, #ff00ee, #00c3ff);
    background: linear-gradient(to top, #ff00ee, #00c3ff); 
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    margin-top: 10vh;
  }

  .aboutContainer {
    display: flex;
    max-width: 100vw;
  
    div {
        display: flex;
        flex-direction: column;
        width: 30vw;
        align-items: center;


      h3 {
        font-size: 2.5em;
        font-family: "KARNIVOT";
        text-align: center;
        background: -webkit-linear-gradient(bottom, #ff00ee, #00c3ff);
        background: linear-gradient(to top, #ff00ee, #00c3ff); 
        -webkit-background-clip: text;
        -webkit-text-fill-color: transparent;
        margin: 6vh 0vw 1vh 0vw;
      }
  
      img {
        height: 40vh;
        width: auto;
        align-self: center;
      }
  
      p {
        width: auto;
        font-size: 0.9em;
        letter-spacing: -0.05em;
        font-family: "joystix";
        color: white;
        background-color: rgba(0, 0, 0, 0.5);
        line-height: 2.4vh;
        text-align: center;
        overflow: normal;
        margin: 1vh 1vw;
      }
    }
  }
}