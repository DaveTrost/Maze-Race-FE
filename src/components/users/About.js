import React from 'react';
import AboutCard from './AboutCard';
import aboutProps from './AboutProps';
import styles from '../styles/About.css';

const About = () => {
  const aboutElements = aboutProps.map(about => {    
    return (
      <li key={about.name + about.img + about.desc}>
        <AboutCard {...about}/>
      </li>
    );
  });

  return (
    <div className={styles.AboutElements}>
      <h2>Meet The Developers</h2>
      <ul>
        {aboutElements}
      </ul>
    </div>
  );
};

export default About;
