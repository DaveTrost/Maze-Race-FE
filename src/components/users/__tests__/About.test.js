import React from 'react';
import { shallow } from 'enzyme';
import About from '../About.js';

describe('About component', () => {
  it('renders About', () => {
    const wrapper = shallow(<About />);
    expect(wrapper).toMatchSnapshot();
  });
});
