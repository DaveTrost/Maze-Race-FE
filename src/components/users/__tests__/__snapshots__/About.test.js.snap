// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders About 1`] = `
<div
  className="AboutElements"
>
  <h2>
    Meet The Developers
  </h2>
  <div
    className="aboutContainer"
  >
    <div
      key="Dylan Corvidaehttps://raw.githubusercontent.com/Team-Amazing-Race/Maze-Race-FE/dev/src/assets/img/dylan.jpgAn empathic and eclectic non-binary software developer, queer circus performer, punk/metal musician, respectful parent & poly partner. My hobbies include: dance trapeze, hand-balancing, connecting to nature, playing and making music on guitar, piano, singing in choirs and screaming in bands."
    >
      <AboutCard
        desc="An empathic and eclectic non-binary software developer, queer circus performer, punk/metal musician, respectful parent & poly partner. My hobbies include: dance trapeze, hand-balancing, connecting to nature, playing and making music on guitar, piano, singing in choirs and screaming in bands."
        image="https://raw.githubusercontent.com/Team-Amazing-Race/Maze-Race-FE/dev/src/assets/img/dylan.jpg"
        name="Dylan Corvidae"
      />
    </div>
    <div
      key="Kamran Alihttps://raw.githubusercontent.com/Team-Amazing-Race/Maze-Race-FE/dev/src/assets/img/kam.jpgMy name is Kam, a full-stack software developer that has proficiency in back-end Javascript node packages like express, and the front-end framework react/redux. I also have a strong background in Digital Design. I am very passionate about clean code and design, because I really want communication between the user and the developers clear."
    >
      <AboutCard
        desc="My name is Kam, a full-stack software developer that has proficiency in back-end Javascript node packages like express, and the front-end framework react/redux. I also have a strong background in Digital Design. I am very passionate about clean code and design, because I really want communication between the user and the developers clear."
        image="https://raw.githubusercontent.com/Team-Amazing-Race/Maze-Race-FE/dev/src/assets/img/kam.jpg"
        name="Kamran Ali"
      />
    </div>
    <div
      key="Sam Jespersenhttps://raw.githubusercontent.com/Team-Amazing-Race/Maze-Race-FE/dev/src/assets/img/sam.gifFull-stack JavaScript developer with a background in cyber-crime investigation and analysis. Driven to create accessible tools that help people accomplish their goals."
    >
      <AboutCard
        desc="Full-stack JavaScript developer with a background in cyber-crime investigation and analysis. Driven to create accessible tools that help people accomplish their goals."
        image="https://raw.githubusercontent.com/Team-Amazing-Race/Maze-Race-FE/dev/src/assets/img/sam.gif"
        name="Sam Jespersen"
      />
    </div>
    <div
      key="Jose Ojedahttps://raw.githubusercontent.com/Team-Amazing-Race/Maze-Race-FE/dev/src/assets/img/jose.jpgSoftware developer who believes good software design always keeps the end-user in mind and is as clean as it is clever. I have a focus on finding client centric solutions that can affect meaningful change rooted in six years experience throughout multiple roles in the insurance industry."
    >
      <AboutCard
        desc="Software developer who believes good software design always keeps the end-user in mind and is as clean as it is clever. I have a focus on finding client centric solutions that can affect meaningful change rooted in six years experience throughout multiple roles in the insurance industry."
        image="https://raw.githubusercontent.com/Team-Amazing-Race/Maze-Race-FE/dev/src/assets/img/jose.jpg"
        name="Jose Ojeda"
      />
    </div>
  </div>
</div>
`;
